boruta/script.whizzml

```diff
@@ -168,24 +210,69 @@
 ;;   fields: (map) Fields information structure
 ;;   f-names: (list) Fields names
 ;;   objective: (string) objective field name
+;;   median-shadows: (number) Median value of the list of maximum shadow 
+;;   importances of all executions
+;;   f-importances: (list) Field importances from all executions
 ;;
 ;; Output:  (map) Map from categories (important, unimportant or undecided) to
 ;;                list of field names
-;;
-(define (group-importance f-rank fields f-names objective)
+(define (group-importance f-rank fields f-names objective median-shadows f-importances)
   (let (to-name (lambda (x) (fields [x "name"] false))
         important (iterate (r [] k (keys f-rank))
                     (if (positive? (f-rank k)) (append r (to-name k)) r))
-        important (append important objective)
+        ;; Objective is also an important field
+        important (append important objective) 
         undecided (iterate (r [] k (keys f-rank))
                     (if (zero? (f-rank k)) (append r (to-name k)) r))
         unimportant (filter (lambda (x)
                               (not (or (member? x important)
                                        (member? x undecided))))
-                            f-names))
-    {"important" important
-     "unimportant" unimportant
-     "undecided" undecided}))
+                            f-names)
+        ;; Obtain id of undecided fields
+        undecided-ids (iterate (r [] k (keys f-rank))
+                        (if (zero? (f-rank k)) (append r k) r))
+        ;; Calculating if undecided fields are important or unimportant
+        undecided-unimportant (undecided-unimportance 
```

Comment:
Not yet! :-) 
```
(undecided-unimportance undecided-ids 
                        median-shadows 
                        f-importances 
                        fields)
```

As to your question, It's my usual practice, but let's wait for jao's opinion about that.